### DISCUSSION TITLE
Building api request for image inference

### DISCUSSION THREAD

**Discussion by BlaiseCz** [Project-MONAI/MONAILabel#349] [Q&A] [answered]
Hello!
I'm trying to build an application that communicates with monai server and uses automatic segmentation. 
Thus I have to make a request to its API. 
Infer curl from localhost:8000/info/  :
 ```
   curl -X 'POST' \
      'http://localhost:8000/infer/pretrained.pt?output=image' \
      -H 'accept: application/json' \
      -H 'Content-Type: multipart/form-data' \
      -F 'params= {"model": "left_atrium", "device": "cpu"}' \
      -F 'file=@la_028.nii.gz;type=application/x-gzip'
```
and now I'm trying to figure out how to paste it into python requests, like this (and many similar versions):
```
import requests

headers = {
    'accept': 'application/json',
    'Content-type': 'Content-Type: multipart/form-data'
}

files = {
    'params': {'model': 'left_atrium', 'device': 'cpu'},
    'file': ('la_027.nii.gz', open('data\heart\la_027.nii.gz', 'rb'), 'application/x-gzip')
}
response = requests.post('http://localhost:8000/infer/pretrained.pt?output=image', headers=headers, files=files)
```
monai server uses 'UploadFile' class from fastApi and it raises ` raise HTTPException(status_code=500, detail="Neither Image nor File input is provided")`
 :(
Do you know how to properly call infer with python requests?

**Comment by SachidanandAlle** [comment, marked as answer]
https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/slicer/MONAILabel/MONAILabelLib/client.py

**Reply by BlaiseCz** [reply]
Thanks for your answer, just found out I can also use request like this: 
`requests.post('http://localhost:8000/infer/left_atrium?image=la_028.nii.gz' , headers=headers, data=data)`
where image parameter can be a path to my nii.gz file. And the final request looks like this:

```
def request_inferenced_image(image, model="left_atrium", device="cuda:0"):
    headers = {
        'accept': 'application/json',
        'Content-type': 'Content-Type: multipart/form-data'
    }

    data = {
        'params': {"model": model, "device": device},
    }

    response = requests.post('http://localhost:8000/infer/left_atrium?image=' + image, headers=headers, data=data)
```
